tests/e2e/flowtest/test_selinux_relabeling_solution.py

```diff
@@ -0,0 +1,255 @@
+import logging
+import os.path
+import yaml
+from shutil import rmtree
+from ocs_ci.ocs.resources import pod
+import time
+from tempfile import mkdtemp, NamedTemporaryFile
+
+from ocs_ci.ocs.resources.pod import wait_for_pods_to_be_running
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import E2ETest, ignore_leftovers
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.helpers import helpers
+
+TARFILE = "cephfs.tar.gz"
+SIZE = "20Gi"
+TFILES = 2000
+SAMPLE_TEXT = b"QWERTYUIOP ASDFGJKL ZXCVBNM"
+ntar_loc = mkdtemp()
+
+log = logging.getLogger(__name__)
+
+
+class TestSelinuxrelabel(E2ETest):
+    def create_files(self):
+        log.info(f"Creating {TFILES} files ")
+        onetenth = TFILES / 10
+        endoften = onetenth - 1
+        tarfile = os.path.join(ntar_loc, TARFILE)
+        one_dir = mkdtemp()
+        log.info(f"dir crated{one_dir}")
+        n = 0
+        while n in range(0, 11):
+            first_dir = mkdtemp(suffix="Fdir", dir=one_dir)
+            log.info(f"created dir {first_dir}")
+            for i in range(0, 11):
+                new_dir = mkdtemp(suffix="Dir", dir=first_dir)
+                log.info(f"Dir created {new_dir}")
+                test_file_list = []
+                for j in range(0, TFILES):
+                    tmpfile = NamedTemporaryFile(dir=new_dir, delete=False)
+                    fname = tmpfile.name
+                    with tmpfile:
+                        tmpfile.write(SAMPLE_TEXT)
+                    if i % onetenth == endoften:
+                        dispv = i + 1
+                        log.info(f"{dispv} local files created")
+                        test_file_list.append(fname.split(os.sep)[-1])
+                first_dir = new_dir
+            n = n + 1
+        run_cmd(f"tar cfz {tarfile} -C {one_dir} .", timeout=1800)
+        rmtree(one_dir)
+        return tarfile
+
+    def create_pvc_pod(self):
+        # Create nfs pvcs with storageclass ocs-storagecluster-ceph-nfs
+        self.namespace = "openshift-storage"
+        self.storage_cluster_obj = ocp.OCP(
+            kind="Storagecluster", namespace=self.namespace
+        )
+        self.pod_obj = ocp.OCP(kind="Pod", namespace=self.namespace)
+        self.pvc_obj = ocp.OCP(kind=constants.PVC, namespace=self.namespace)
+        self.pv_obj = ocp.OCP(kind=constants.PV, namespace=self.namespace)
+        self.nfs_sc = "ocs-storagecluster-cephfs"
+
+        self.nfs_pvc_obj = helpers.create_pvc(
+            sc_name=self.nfs_sc,
+            namespace=self.namespace,
+            pvc_name="nfs-pvc-2",
+            size="10Gi",
+            do_reload=True,
+            access_mode=constants.ACCESS_MODE_RWO,
+            volume_mode="Filesystem",
+        )
+        log.info(f"{self.nfs_pvc_obj} created")
+
+        # Create deployment config for app pod
+        log.info("----create deployment config----")
+        deployment_config = """
```

Comment:
since this config is static, probably will be better to use a common approach to store it like a yaml file, to reuse afterwards. WDYT?